crypto/param_build.c

```diff
@@ -159,6 +159,33 @@ int ossl_param_bld_push_BN(OSSL_PARAM_BLD *bld, const char *key,
     return 1;
 }
 
+int ossl_param_bld_push_BN_pad(OSSL_PARAM_BLD *bld, const char *key,
+                               const BIGNUM *bn, size_t sz)
+{
+    int n, secure = 0;
+    OSSL_PARAM_BLD_DEF *pd;
+
+    if (bn != NULL) {
+        n = BN_num_bytes(bn);
+        if (n < 0) {
```

Comment:
Is n = 0 ok?

Reply:
Yes.  Zero can have zero size I think.

Reply:
A padded BN that can only handle zero seems not useful.  I would make zero-size an error.

Reply:
I'd expect `ossl_param_bld_push_BN_pad(&bld, "bn", bn, BN_num_bytes(bn))` to always work, this suggestion would prevent that.

Reply:
I would not expect to make a call like that.  I would expect to compare a couple of BIGNUM's and use the same size padding for all BN params.  Why have a "padded" type if you're going to leak the size of zero?